pkg/envoy/agent.go

```diff
@@ -148,17 +151,27 @@ func (a *Agent) terminate() {
 		time.Sleep(a.minDrainDuration)
 		log.Infof("Checking for active connections...")
 		ticker := time.NewTicker(activeConnectionCheckDelay)
-		for range ticker.C {
-			if a.activeProxyConnections() == 0 {
+		podTerminationGracePeriod, _ := time.ParseDuration(podTerminationDurationVar.Get() + "s")
+		select {
+		case <-ticker.C:
+			ac := a.activeProxyConnections()
+			if ac == 0 {
 				log.Info("There are no more active connections. terminating proxy...")
 				a.abortCh <- errAbort
 				return
 			}
+			log.Infof("There are still %d connections active", ac)
+		// This is needed for cases where Jobs's proxy is not terminated by Kubernetes
+		// and Job kill the proxy using agent's /quitquitquit endpoint.
+		case <-time.After(podTerminationGracePeriod):
```

Comment:
why we need this, after `podTerminationGracePeriod`, if this process exist, kubelet will send kill -9 to it.

Reply:
Please see the comment above ^^. For Jobs it is not doing it.

Reply:
ic

Reply:
After job completes, who will terminate the sidecar?

Reply:
Job triggers `/quitquitquit` endpoint after it finishes.